### PR TITLE
small optimization in eval_bit in univariate_poly

### PR DESCRIPTION
**Pull request by srajangarg**
https://github.com/symengine/symengine/pull/865#discussion_r56625981

### CODE REVIEW

File: symengine/polynomial.cpp [hunk 1]
```diff
@@ -160,16 +160,15 @@ integer_class UnivariateIntPolynomial::eval(const integer_class &x) const {
 integer_class UnivariateIntPolynomial::eval_bit(const int &x) const {
 
     unsigned int last_deg = dict_.rbegin()->first;
-    integer_class result(0), two_pow;
+    integer_class result(0), two_pow_result;
 
     for (auto it = dict_.rbegin(); it != dict_.rend(); ++it) {
 
-        two_pow = integer_class(1) << x * (last_deg - (*it).first);
+        two_pow_result = result << x * (last_deg - (*it).first);
```
Comment (isuruf) [inline symengine/polynomial.cpp]: You don't need the temporary. Operation can be done on `result` itself

**Comment by Sumith1896** [comment]
Fine by me, +1

File: symengine/polynomial.cpp [hunk 2]
```diff
@@ -160,16 +160,14 @@ integer_class UnivariateIntPolynomial::eval(const integer_class &x) const {
 integer_class UnivariateIntPolynomial::eval_bit(const int &x) const {
 
     unsigned int last_deg = dict_.rbegin()->first;
-    integer_class result(0), two_pow;
+    integer_class result(0);
 
     for (auto it = dict_.rbegin(); it != dict_.rend(); ++it) {
 
-        two_pow = integer_class(1) << x * (last_deg - (*it).first);
+        result = (*it).second + (result << x * (last_deg - (*it).first));
```
Comment (isuruf) [inline symengine/polynomial.cpp]: This is a good use case to check whether https://github.com/symengine/symengine/issues/478#issuecomment-153172972 would have a significant performance improvement when `-DINTEGER_CLASS=gmp` is given

**Comment by Sumith1896** [comment]
Thanks @srajangarg

**Pull request merged**